modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -1234,6 +1235,14 @@ private void processOperation(String resourcePath, String httpMethod, Operation
         final List<SecurityRequirement> globalSecurities = openAPI.getSecurity();
         for (Tag tag : tags) {
             try {
+                // Replacing reference by corresponding ApiResponse
+                for(String key : operation.getResponses().keySet()) {
+                    ApiResponse apiResponse = operation.getResponses().get(key);
+                    if(apiResponse.get$ref() != null) {
+                        operation.getResponses().put(key, ModelUtils.getReferencedApiResponse(openAPI, apiResponse));
+                    }
+                }
+
```

Comment:
@axel7083 thanks for the PR. Should the fix be done in the default codegen to get the referenced response object/schema correctly?

Reply:
Btw, I think I might have fixed it in the latest master.

Can you please give it a try with the latest master?

Reply:
Hey, sadly I get the same error.

>   Exception: Cannot invoke "io.swagger.v3.oas.models.media.Content.get(Object)" because the return value of "io.swagger.v3.oas.models.responses.ApiResponse.getContent()" is null

	at org.openapitools.codegen.DefaultGenerator.processOperation(DefaultGenerator.java:1340)
	at org.openapitools.codegen.DefaultGenerator.processPaths(DefaultGenerator.java:1234)
	at org.openapitools.codegen.DefaultGenerator.generateApis(DefaultGenerator.java:608)
	at org.openapitools.codegen.DefaultGenerator.generate(DefaultGenerator.java:1065)
	at org.openapitools.codegen.java.spring.SpringCodegenTest.testSSEOperationSupport(SpringCodegenTest.java:3235)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:577)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:135)
	at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:673)
	at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:220)
	at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
	at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:945)
	at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:193)
	at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.testng.TestRunner.privateRun(TestRunner.java:808)
	at org.testng.TestRunner.run(TestRunner.java:603)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:429)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:423)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:383)
	at org.testng.SuiteRunner.run(SuiteRunner.java:326)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1249)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
	at org.testng.TestNG.runSuites(TestNG.java:1092)
	at org.testng.TestNG.run(TestNG.java:1060)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:105)
Caused by: java.lang.NullPointerException: Cannot invoke "io.swagger.v3.oas.models.media.Content.get(Object)" because the return value of "io.swagger.v3.oas.models.responses.ApiResponse.getContent()" is null
	at org.openapitools.codegen.languages.SpringCodegen.lambda$fromOperation$16(SpringCodegen.java:1280)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at org.openapitools.codegen.languages.SpringCodegen.fromOperation(SpringCodegen.java:1282)
	at org.openapitools.codegen.DefaultGenerator.processOperation(DefaultGenerator.java:1308)
	... 29 more

Reply:
You can find in https://github.com/OpenAPITools/openapi-generator/blob/2129b759a31bc1b22e84728053f5fed7fe137990/modules/openapi-generator/src/test/resources/3_0/sse.yaml the file including the failing test on the latest master

Reply:
did a test with the following using the latest master
```
java -jar modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -g spring -i https://raw.githubusercontent.com/OpenAPITools/openapi-generator/2129b759a31bc1b22e84728053f5fed7fe137990/modules/openapi-generator/src/test/resources/3_0/sse.yaml -o /tmp/spring -c /tmp/config.json
```
and the output compiles without issues

Reply:
the snapshot version of the latest master can be found in https://oss.sonatype.org/content/repositories/snapshots/org/openapitools/openapi-generator-cli/7.2.0-SNAPSHOT/

Reply:
happy to reopen this PR if needed. closing it for the time being

Reply:
I cannot reopen